xarray/core/dataset.py

```diff
@@ -521,6 +551,12 @@ def __setitem__(self, key, value):
         if utils.is_dict_like(key):
             raise NotImplementedError('cannot yet use a dictionary as a key '
                                       'to set Dataset values')
+
+        if key in self._level_coords:
+            raise ValueError("%r is already a MultiIndex level of "
```

Comment:
Just checked and confirm that this is already checked by `update`.

I'd be OK to remove this. Maybe it is a good thing not to have too many different messages for similar errors...